R/runner.R

```diff
@@ -314,13 +314,15 @@ orderly_runner_ <- R6::R6Class(
       redis_key <- workflow_redis_key(self$queue$queue_id, workflow_key)
       self$con$SADD(self$keys$key_workflows, workflow_key)
       self$con$SADD(redis_key, report_keys)
+      reports <- data.frame(key = report_keys,
+                            execution_order = seq_along(report_keys))
       ## Return in same order we received the reports
       ## These will be used by OW to map returned ID to specific report
       return_order <- vnapply(workflow, "[[", "original_order")
-      report_keys <- report_keys[order(return_order)]
+      reports <- reports[order(return_order), ]
```

Comment:
It looks here that we have two orders we want to retain: input order and execution order but we communicate this via ordering of the array and a key within the array. Might it be better to expand the inner object to include `original_order` as a key too?